src/main/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/OrganisationService.java

```diff
@@ -0,0 +1,206 @@
+                               BatchLastRunTimestampRepository batchLastRunTimestampRepository,
+                               DatabaseDateTimeRepository databaseDateTimeRepository,
+                               ProcessEventTracker processEventTracker,
+                               @Value("${groupAccess.lastRunTimeTolerance}") String tolerance
+                               ) {
+        this.prdService = prdService;
+        this.profileRefreshQueueRepository = profileRefreshQueueRepository;
+        this.organisationRefreshQueueRepository = organisationRefreshQueueRepository;
+        this.accessTypesRepository = accessTypesRepository;
+        this.batchLastRunTimestampRepository = batchLastRunTimestampRepository;
+        this.databaseDateTimeRepository = databaseDateTimeRepository;
+        this.pageSize = pageSize;
+        this.jdbcTemplate = jdbcTemplate;
+        this.processEventTracker = processEventTracker;
+        this.tolerance = tolerance;
+    }
+
+    @Transactional
+    public void findOrganisationChangesAndInsertIntoOrganisationRefreshQueue() {
+        log.info("findOrganisationChangesAndInsertIntoOrganisationRefreshQueue started...");
+        ProcessMonitorDto processMonitorDto = new ProcessMonitorDto("PRM Process 3 - Find organisation changes");
+        processEventTracker.trackEventStarted(processMonitorDto);
+
+        try {
+            final DatabaseDateTime batchRunStartTime = databaseDateTimeRepository.getCurrentTimeStamp();
+            List<AccessTypesEntity> allAccessTypes = accessTypesRepository.findAll();
+            if (allAccessTypes.size() != 1) {
+                throw new ServiceException("Single AccessTypesEntity not found");
+            }
+            AccessTypesEntity accessTypesEntity = allAccessTypes.get(0);
+            List<BatchLastRunTimestampEntity> allBatchLastRunTimestampEntities = batchLastRunTimestampRepository
+                    .findAll();
+            if (allBatchLastRunTimestampEntities.size() != 1) {
+                throw new ServiceException("Single BatchLastRunTimestampEntity not found");
+            }
+            BatchLastRunTimestampEntity batchLastRunTimestampEntity = allBatchLastRunTimestampEntities.get(0);
+            LocalDateTime orgLastBatchRunTime = batchLastRunTimestampEntity.getLastOrganisationRunDatetime();
+
+            int toleranceSeconds = Integer.parseInt(tolerance);
+            LocalDateTime sinceTime = orgLastBatchRunTime.minusSeconds(toleranceSeconds);
+            String formattedSince = ISO_DATE_TIME_FORMATTER.format(sinceTime);
+
+            Integer accessTypeMinVersion = accessTypesEntity.getVersion().intValue();
+            OrganisationsResponse organisationsResponse = prdService
+                    .retrieveOrganisations(formattedSince, 1, Integer.valueOf(pageSize)).getBody();
+            writeAllToOrganisationRefreshQueue(organisationsResponse, accessTypeMinVersion);
+
+            int page = 2;
```

Comment:
a do/while loop might be better suited here if you know you have to call the api at least once?